Yoast/wordpress-seo
# Only trim the meta description if the text has space

## Context


* This PR is to fix [this issue](https://github.com/Yoast/wordpress-seo/issues/17116).

## Summary


This PR can be summarized in the following changelog entry:

* Fixes the bug where the meta description gets trimmed to an empty string if the text doesn't have any space.

## Relevant technical choices:

* When the text doesn't have any space, the first 156 chars from the content will be retrieved as the meta description and wouldn't be trimmed further.
* For now, this fix will be released behind "JAPANESE_SUPPORT" feature flag.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* In PHPStorm/your preferred IDE, navigate to `packages/yoastseo/src/languageProcessing/languages/_default` and copy the contents
* If `ja` folder is not available yet, inside the `languages` folder, create a `ja` folder and paste the contents from `_default` (don't commit any changes)
* Build the Free plugin
* Enable the `JAPANESE_SUPPORT` feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', true );` in `basic-wordpress-config.php` file
* Go to "SEO > Search Appearance".
* Set any post / page / CPTS' "Meta description" to excerpt by using %%excerpt%%. Save changes.
* Set your site to Japanese` 日本語`
* Make a new post / page / CPTS
* Confirm that the Meta Description is set to `excerpt`
<img width="599" alt="Screenshot 2021-06-11 at 11 44 15" src="https://user-images.githubusercontent.com/48715883/121667366-7d472e00-caaa-11eb-9646-b134cc3e84a2.png">

* Copy and paste the Japanese text below. Don't add anything else to the Meta Description box. Publish.
ネルソン君はそう受売でしかるばならんないのましですない。（するとかたがたでし以上ないないませてあるはなりですませて、）もともとするなけれ道具に、Englandの主義だけするて云いという、魚籃の永続も昨日の以上なり突き破っもっ事と執っなば答弁ら重んずるが得なといった皆洋服た事た。あなたはあたかも自分が進まですように飽いて切らでつもりないてそれでそれほど目黒人したう。

* Check in Yoast metabox in the editor
* Confirm that the Google Preview returns this result:
<img width="605" alt="Screenshot 2021-06-11 at 11 48 12" src="https://user-images.githubusercontent.com/48715883/121667948-15451780-caab-11eb-96aa-9935ce6b169d.png">

* Open the post / page / CPTS in the front page
* Open the source code
* Confirm that the content of meta description tag `<meta name="description" content="">` does not say  "No description has been set", or something along those lines.


* Disable the `JAPANESE_SUPPORT` by adding this  `define( 'YOAST_SEO_JAPANESE_SUPPORT', false );` in `basic-wordpress-config.php` file
* Reload your page
* Confirm that the Google Preview doesn't show any excerpt from the text. Instead it should show a text that asks you to set a meta description.
<img width="603" alt="Screenshot 2021-09-09 at 14 34 44" src="https://user-images.githubusercontent.com/48715883/132687303-c1980af3-e036-4f33-872e-13232df5499c.png">

* Open the post / page / CPTS in the front page
* Open the source code
* Confirm that you cannot find the meta description tag by searching `<meta name="description" content`

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* Install and activate Free plugin
* Enable the `JAPANESE_SUPPORT` feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', true );` in `basic-wordpress-config.php` file
* Go to "SEO > Search Appearance".
* Set any post / page / CPTS' "Meta description" to excerpt by using %%excerpt%%. Save changes.
* Set your site to Japanese` 日本語`
* Make a new post / page / CPTS
* Confirm that the Meta Description is set to `excerpt`
<img width="599" alt="Screenshot 2021-06-11 at 11 44 15" src="https://user-images.githubusercontent.com/48715883/121667366-7d472e00-caaa-11eb-9646-b134cc3e84a2.png">

* Copy and paste the Japanese text below. Don't add anything else to the Meta Description box. Publish.
ネルソン君はそう受売でしかるばならんないのましですない。（するとかたがたでし以上ないないませてあるはなりですませて、）もともとするなけれ道具に、Englandの主義だけするて云いという、魚籃の永続も昨日の以上なり突き破っもっ事と執っなば答弁ら重んずるが得なといった皆洋服た事た。あなたはあたかも自分が進まですように飽いて切らでつもりないてそれでそれほど目黒人したう。

* Check in Yoast metabox in the editor
* Confirm that the Google Preview returns this result:
<img width="605" alt="Screenshot 2021-06-11 at 11 48 12" src="https://user-images.githubusercontent.com/48715883/121667948-15451780-caab-11eb-96aa-9935ce6b169d.png">

* Open the post / page / CPTS in the front page
* Open the source code
* Confirm that the content of meta description tag `<meta name="description" content="">` does not say  "No description has been set", or something along those lines.


* Disable the `JAPANESE_SUPPORT` by adding this  `define( 'YOAST_SEO_JAPANESE_SUPPORT', false );` in `basic-wordpress-config.php` file
* Reload your page
* Confirm that the Google Preview doesn't show any excerpt from the text. Instead it should show a text that asks you to set a meta description.
<img width="603" alt="Screenshot 2021-09-09 at 14 34 44" src="https://user-images.githubusercontent.com/48715883/132687303-c1980af3-e036-4f33-872e-13232df5499c.png">

* Open the post / page / CPTS in the front page
* Open the source code
* Confirm that you cannot find the meta description tag by searching `<meta name="description" content`

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-864
